specification/cognitiveservices/OpenAI.Inference/models/completions.common.tsp

```diff
@@ -35,3 +37,56 @@ enum CompletionsFinishReason {
     """)
   contentFiltered: "content_filter",
 }
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Ratings for the intensity and risk level of harmful content.")
+enum ContentFilterSeverity {
+  @doc("General content or related content in generic or non-harmful contexts.")
+  safe: "safe";
+
+  @doc("Harmful content at a low intensity and risk level.")
+  low: "low";
+
+  @doc("Harmful content at a medium intensity and risk level.")
+  medium: "medium";
+
+  @doc("Harmful content at a high intensity and risk level.")
+  high: "high";
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about filtered content severity level and if it has been filtered or not.")
+model ContentFilterResult {
+  @doc("")
+  @projectedName("json", "severity")
+  severity: ContentFilterSeverity;
+
+  @doc("")
+  @projectedName("json", "filtered")
+  filtered: boolean;
+
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about the content filtering category, if it has been detected.")
+model ContentFilterResults {
+  sexual: ContentFilterResult;
+  violence: ContentFilterResult;
+  hate: ContentFilterResult;
+  selfHarm: ContentFilterResult;
+  //error?
```

Comment:
`error` is a listed field in the Swagger schema but do I add it here? Didn't seem obvious from looking at other objects

Reply:
I think you probably don't have to add error here. That would be a variant that will be taken care for you. Unless it's a custom thing. `error` will be a variant of the possible response of the HTTP method anyway.

Reply:
How do I discern which properties of an object defined in Swagger need to be authored in TypeSpec?

Looking at [the schema for `contentFilterResults`](https://github.com/Azure/azure-rest-api-specs/blob/32b369c929d4aad3bbe4d428c9c434186e52552a/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-06-01-preview/inference.json#L1007C16-L1007C16), each property is only a reference to another schema definition.

Reply:
That's a tough one, I personally don't have a rule of thumb for that. Generally, anything with documentation that is clearly related to the domain, is probably something that you need to reflect in TSP. In this Particular case, it does seem like the structure of `ErrorBase` is being re-used in the one you linked. 

It does seem like that property is present on every response. If you are able to send out a request with the generated code, I would try that and see what json you receive.